File: JavaScript/src/LitJson/IJsonWrapper.js
import { JsonType } from './JsonType.js';

class IJsonWrapper {
    get IsArray() {
        throw new Error('Not implemented');
    }

    get IsBoolean() {
        throw new Error('Not implemented');
    }

    get IsDouble() {
        throw new Error('Not implemented');
    }

    get IsObject() {
        throw new Error('Not implemented');
    }

    get IsString() {
        throw new Error('Not implemented');
    }

    get IsNull() {
        throw new Error('Not implemented');
    }

    SetBoolean(val) {
        throw new Error('Not implemented');
    }

    SetDouble(val) {
        throw new Error('Not implemented');
    }

    SetJsonType(type) {
        throw new Error('Not implemented');
    }

    SetString(val) {
        throw new Error('Not implemented');
    }

    SetNull() {
        throw new Error('Not implemented');
    }

    Add(val) {
        throw new Error('Not implemented');
    }

    Set(key, val) {
        throw new Error('Not implemented');
    }
}

export { IJsonWrapper };

File: java1/src/main/java/toolgood/algorithm/litJson/IJsonWrapper.java
package toolgood.algorithm.litJson;

import java.math.BigDecimal;

public interface IJsonWrapper {
    
    boolean IsArray();
    boolean IsBoolean();
    boolean IsDouble();
    boolean IsObject();
    boolean IsString();
    boolean IsNull();

    void SetBoolean(boolean val);
    void SetDouble(BigDecimal val);
    void SetJsonType(JsonType type);
    void SetString(String val);
    void SetNull();

    void Add(IJsonWrapper val);

    void Set(String key, IJsonWrapper val);


}